pkg/log/log.go

```diff
@@ -165,7 +164,17 @@ func format(msg, level string, ts time.Time, appInfo Marshaler, mm Many) string
 
 	var b bytes.Buffer
 	if err := json.NewEncoder(&b).Encode(entry); err != nil {
-		log.Fatal(err)
+		// at this point we need to report the serialization error.
+		// do it in a JSON object so parsers have a better chance of understanding it
+		err = json.NewEncoder(&b).Encode(map[string]string{
+			"message": fmt.Sprintf("gobox/log: failed to JSON encode log entry %v of type %T; err=%v", entry,
```

Comment:
How much do we rely on the log serializer to omit sensitive data? It's pretty worrying to serialize an entire object blindly, given that it could contain sensitive info.

What about just logging the type and the error?

Reply:
into it

Reply:
fixed

Reply:
Great catch.